### PR TITLE
[FLINK-12189][Runtime]Fix bugs and typos in memory management related…

### PR DESCRIPTION
**Pull request by liyafan82**
… code

## What is the purpose of the change

Fix some bugs and typos in the memory related source code. For example,

1. In MemoryManager.release method, it should throw an NPE, if the input is null. But it does not. 
2. In HybridMemorySegment.put and get methods, the number of bytes read/written should be specified by the input parameter, rather than reading/writing until the end.
3. In HeapMemorySegment, the data member memory should be returned. This is because the member heapArray will not be reset to null after calling the release method.

## Brief change log

Change related classes in flink-runtime

## Verifying this change

This change is a trivial rework / code cleanup without any test coverage.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (no)
  - The serializers: (no)
  - The runtime per-record code paths (performance sensitive): (no)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Yarn/Mesos, ZooKeeper: (no)
  - The S3 file system connector: (no)

## Documentation

  - Does this pull request introduce a new feature? (no)

### CODE REVIEW

File: flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java [hunk 1]
```diff
@@ -387,11 +387,11 @@ public void release(MemorySegment segment) {
...
 	 * @param segments The segments to be released.
 	 * @throws NullPointerException Thrown, if the given collection is null.
-	 * @throws IllegalArgumentException Thrown, id the segments are of an incompatible type.
+	 * @throws IllegalArgumentException Thrown, if the segments are of an incompatible type.
 	 */
 	public void release(Collection<MemorySegment> segments) {
 		if (segments == null) {
-			return;
+			throw new NullPointerException();
 		}
```
Comment (chunweilei) [inline flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java]: Why should it throw NullPointerException? Besides, I think a test case is needed.

Comment (liyafan82) [reply flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java]: > Why should it throw NullPointerException? Besides, I think a test case is needed if it should throw NullPointerException.

Hi chunweilei, thanks a lot for your comments. The Javadoc for this method says "@throws NullPointerException Thrown, if the given collection is null."
So an NPE should be thrown, or this comment should be removed.
According to the style of the other part of the code, I think throwing an NPE is preferred.

Comment (chunweilei) [reply flink-runtime/src/main/java/org/apache/flink/runtime/memory/MemoryManager.java]: Sounds reasonable. Thanks for your kindly reply.

File: flink-core/src/main/java/org/apache/flink/core/memory/MemorySegment.java [hunk 2]
```diff
@@ -1272,7 +1272,7 @@ else if (target.address > target.addressLimit) {
 
 	/**
 	 * Bulk copy method. Copies {@code numBytes} bytes to target unsafe object and pointer.
-	 * NOTE: This is a unsafe method, no check here, please be carefully.
+	 * NOTE: This is a unsafe method, no check here, please be careful.
```
Comment (synckey) [inline flink-core/src/main/java/org/apache/flink/core/memory/MemorySegment.java]: `a` -> `an`?

Comment (liyafan82) [reply flink-core/src/main/java/org/apache/flink/core/memory/MemorySegment.java]: Thanks for the good cache. This PR is about to be closed. Please look at here:
https://github.com/apache/flink/pull/8408

**Pull request closed without merging**